app/code/Magento/InventoryExportStock/Model/GetExportStockData.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryExportStock\Model;
+
+use Exception;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Api\SearchResultsInterface;
+use Magento\InventoryCatalogApi\Api\DefaultStockProviderInterface;
+use Magento\InventoryExportStockApi\Api\Data\ExportStockDataSearchResultInterface;
+use Magento\InventoryExportStockApi\Api\Data\ExportStockDataSearchResultInterfaceFactory;
+use Magento\InventorySalesApi\Api\GetProductSalableQtyInterface;
+
+/**
+ * Class GetExportStockData
+ */
+class GetExportStockData
+{
+    /**
+     * @var ProductRepositoryInterface
+     */
+    private $productRepository;
+
+    /**
+     * @var ExportStockDataSearchResultInterfaceFactory
+     */
+    private $exportStockDataSearchResultFactory;
+
+    /**
+     * @var GetProductSalableQtyInterface
+     */
+    private $getProductSalableQty;
+
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
+
+    /**
+     * ExportStockData constructor
+     *
+     * @param ProductRepositoryInterface $productRepository
+     * @param ExportStockDataSearchResultInterfaceFactory $exportStockDataSearchResultFactory
+     * @param GetProductSalableQtyInterface $getProductSalableQty
+     * @param DefaultStockProviderInterface $defaultStockProvider
+     */
+    public function __construct(
+        ProductRepositoryInterface $productRepository,
+        ExportStockDataSearchResultInterfaceFactory $exportStockDataSearchResultFactory,
+        GetProductSalableQtyInterface $getProductSalableQty,
+        DefaultStockProviderInterface $defaultStockProvider
+    ) {
+        $this->productRepository = $productRepository;
+        $this->exportStockDataSearchResultFactory = $exportStockDataSearchResultFactory;
+        $this->getProductSalableQty = $getProductSalableQty;
+        $this->defaultStockProvider = $defaultStockProvider;
+    }
+
+    /**
+     * Provides product stock data according to search criteria
+     *
+     * @param SearchCriteriaInterface $searchCriteria
+     * @param int $stockId
+     * @param int $qtyForNotManageStock
+     * @return ExportStockDataSearchResultInterface
+     * @throws Exception
```

Comment:
a low-level exception should not be thrown out of high-level service contracts, just a successor of Localized Exception could be thrown at this level